src/api/app/controllers/webui/request_controller.rb

```diff
@@ -244,10 +241,9 @@ def diff
   def list
     redirect_to user_show_path(User.current) and return unless request.xhr? # non ajax request
     requests = BsRequest.list_ids(params)
-    elide_len = 44
-    elide_len = params[:elide_len].to_i if params[:elide_len]
-    session[:requests] = requests
-    requests = BsRequest.collection(ids: session[:requests])
+    elide_len = (params[:elide_len] || 44).to_i
+    session[:request_numbers] = requests.map { |id| BsRequest.find(id).number }.uniq
+    requests = BsRequest.collection(ids: requests)
```

Comment:
I almost think we could replace using ids by using numbers here. But I won't be around next week. So no time to dive deeper.